src/core/currency.js

```diff
@@ -1,277 +1,58 @@
 'use strict';
 const _ = require('lodash');
-const assert = require('assert');
-const utils = require('./utils');
-const BN = require('bn.js');
 
-function Currency() {
-  // Internal form: 0 = XRP. 3 letter-code.
-  // XXX Internal should be 0 or hex with three letter annotation when valid.
-
-  // Json form:
-  //  '', 'XRP', '0': 0
-  //  3-letter code: ...
-  // XXX Should support hex, C++ doesn't currently allow it.
-
-  this._value = NaN;
-  this._update();
+function isISOCode(currency) {
+  return /^[A-Z0-9]{3}$/.test(currency);
```

Comment:
Yeah, I remember that. Those aren't actually ISO codes though, so if we support them we would have to rename the function. My preference would be to just leave it as is and try to discourage people from using non-ISO 3-character codes.